src/GraphQL/Resolver.hs

```diff
@@ -326,3 +330,98 @@ instance forall typeName interfaces fields m.
     case obj of
       Nothing -> pure (Result [DuplicateFields r] GValue.ValueNull)
       Just object -> pure (Result errs (GValue.ValueObject object))
+
+
+-- TODO(tom): we're getting to a point where it might make sense to
+-- split resolver into submodules (GraphQL.Resolver.Union  etc.)
```

Comment:
+1. Or even GraphQL.Internal.Resolver.Union, and have GraphQL.Resolver just re-export the public facing bits.